lib/marshalingCommands.js

```diff
@@ -0,0 +1,173 @@
+var _ = require('lodash');
+var unexpectedItemHandler = require('./unexpectedItemHandler');
+
+/**
+ * Converts boolean value to DynamoDb "BOOL"
+ *
+ * @param item
+ * @returns {*}
+ */
+function marshalBoolean(item) {
+  if (!_.isBoolean(item)) {
```

Comment:
Sorry, it was early in the morning and my thinking was a bit fuzzy. What I meant was this:

```js
function marshall(predicate, transform, item) {
  return predicate(item) ? transform(item) : void 0;
}

function marshalBoolean(item) {
  return marshall(_.isBoolean, function(item) {
        return {BOOL: item};
    }, item);
}
```

You could reduce the boilerplate further if you curry `marshall`:

```js
var marshall = _.curry(function(predicate, transform, item) {
  return predicate(item) ? transform(item) : void 0;
});

var marshalBoolean = marshall(_.isBoolean, function(item) {
    return {BOOL: item};
});
```

You could probably remove that anonymous function if you use something like `lodash-fp`, which I normally wouldn't recommend because [Ramda](http://ramdajs.com) is far better but given the limited scope it's probably fine. In Ramda it would be:

```js
var marshall = marshall(R.is(Boolean), R.objOf('BOOL'));
```
I presume you can do something similar with `lodash-fp` but I just don't know the API.